### PR TITLE
Updated building instructions for Linux

### PR DESCRIPTION
**Pull request by DarkMatter-999**
Added updated instructions for building on Linux and added a script to auto-download the correct bazel version and add it to local venv path.

### CODE REVIEW

**Review by j20001970 — COMMENTED** [review]
It's a very good idea to download bazel automatically to python venv imo, but I think there are several points that can be improved before we get this merged:

1. macOS and Windows could also experience similar problems from using different bazel version, we could address this altogether by converting the download step to python script, and make it part of `setup.py` while creating venv.
2. Since we are using pre-compiled bazel anyways, we could download bazelisk instead to automatically select required version for `mediapipe` submodule, as they could update required bazel version in the future.

How do you think about the improvements above?

**Comment by DarkMatter-999** [comment]
Yeah sure converting to python script and integrating it into the setup process itself is indeed a very good idea.

Just needed a small clarification how do you plan to handle the downloads?
1. We can do it with Python itself using the `urlrlib`
2. Let the platform handle it like `wget`/`curl` for Linux and MacOS and `iwr` for Windows.

**Comment by j20001970** [comment]
> Yeah sure converting to python script and integrating it into the setup process itself is indeed a very good idea.
> 
> Just needed a small clarification how do you plan to handle the downloads?
> 
>     1. We can do it with Python itself using the `urlrlib`
> 
>     2. Let the platform handle it like `wget`/`curl` for Linux and MacOS and `iwr` for Windows.

Downloading through Python itself would be more suitable, so that we can maintain one code path for this task.

**Comment by DarkMatter-999** [comment]
Added auto-download of the latest bazelisk version to the `setup.py` itself.  ✌️

Is there any need to update the `requirements.txt` ?

File: docs/BUILDING.md [hunk 1]
```diff
@@ -55,6 +55,8 @@ Run `build.py --help` to see all options.
 1. Install OpenCV and FFmpeg, then modify `mediapipe/third_party/opencv_linux.BUILD` to make OpenCV visible to Bazel.
 2. Run `build.py desktop` to build desktop library.
 
+NOTE:- Bazel version 6.1.1 is strictly required and is installed automatically by running the `setup.py`, it is installed inside the Python venv.
```
Comment (j20001970) [inline docs/BUILDING.md:58]: Nitpicking: No need to state required Bazel version explicitly if setup script has already handled for us, or we would need to update this section every time mediapipe change required version.

**Comment by j20001970** [comment]
> Is there any need to update the `requirements.txt` ?

I prefer leaving it unchanged unless there are known issues or package deprecation.

File: docs/BUILDING.md (hunk 1 above)
Comment (DarkMatter-999) [reply docs/BUILDING.md:58]: So.... I think I should just remove this whole line altogether

**Comment by j20001970** [comment]
Thanks!

**Pull request merged**